File: January 2026/08.py
from collections import defaultdict

class Solution:
    def countSubarrays(self, arr, k):
        count = 0
        odd_count = 0
        freq = defaultdict(int)
        freq[0] = 1  # Base case: zero odd numbers seen

        for num in arr:
            if num % 2 != 0:
                odd_count += 1

            # If we've seen (odd_count - k) before, it means there's a subarray ending here with k odds
            count += freq[odd_count - k]
            freq[odd_count] += 1

        return count

File: January 2026/08.c
#include <stdio.h>

#define MAXN 100005

int countSubarrays(int arr[], int n, int k) {
    int freq[MAXN] = {0};
    freq[0] = 1;  // Base case
    int oddCount = 0, result = 0;

    for (int i = 0; i < n; ++i) {
        if (arr[i] % 2 != 0)
            oddCount++;

        if (oddCount >= k)
            result += freq[oddCount - k];

        freq[oddCount]++;
    }

    return result;
}

int main() {
    int arr[] = {2, 2, 5, 6, 9, 2, 11};
    int k = 2;
    int n = sizeof(arr) / sizeof(arr[0]);
    printf("%d\n", countSubarrays(arr, n, k));  // Output: 8
    return 0;
}
